src/Features/Core/Portable/ExternalAccess/UnitTesting/UnitTestingSolutionCrawlerServiceAccessor.cs

```diff
@@ -12,13 +12,47 @@ namespace Microsoft.CodeAnalysis.ExternalAccess.UnitTesting
 {
     internal sealed class UnitTestingSolutionCrawlerServiceAccessor : IUnitTestingSolutionCrawlerServiceAccessor
     {
-        private readonly ISolutionCrawlerService _implementation;
+        private readonly ISolutionCrawlerRegistrationService _registrationService;
+        private readonly ISolutionCrawlerService _solutionCrawlerService;
+
+        private UnitTestingIncrementalAnalyzerProvider _analyzerProvider;
 
         [Obsolete(MefConstruction.FactoryMethodMessage, error: true)]
-        public UnitTestingSolutionCrawlerServiceAccessor(ISolutionCrawlerService implementation)
-            => _implementation = implementation;
+        public UnitTestingSolutionCrawlerServiceAccessor(
+            ISolutionCrawlerRegistrationService registrationService,
+            ISolutionCrawlerService solutionCrawlerService)
+        {
+            _registrationService = registrationService;
+            _solutionCrawlerService = solutionCrawlerService;
+        }
+
+        public void AddAnalyzerProvider(IUnitTestingIncrementalAnalyzerProviderImplementation provider, UnitTestingIncrementalAnalyzerProviderMetadataWrapper metadata)
```

Comment:
😕 Why not just import `IUnitTestingIncrementalAnalyzerProviderImplementation` in the constructor?

Reply:
at the moment IUnitTestingIncrementalAnalyzerProviderImplementation implementation is not a MEF component. @shyamnamboodiripad might have more insight as to why that is

Reply:
I think its because we don't want an open of the TW (which is what initializes test discovery) to load Roslyn binaries if the solution does not include any C# / VB code. Currently after TW is opened, we wait for UIContexts to tell us that the solution contains C# / VB projects and only register the analyzer if this condition is satisfied.

Reply:
I see. The issue, if any, would be the other way around, since this assembly isn't loaded unless we already know a C# or VB project is needed.